metrics/web/test/common/data/model/build_day_data_test.dart

```diff
@@ -0,0 +1,335 @@
+            projectId: projectId,
+            successful: successful,
+            failed: failed,
+            unknown: unknown,
+            inProgress: inProgress,
+            totalDuration: totalDuration,
+            day: null,
+          ),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      "creates an instance with the given parameters",
+      () {
+        final buildDay = BuildDayData(
+          projectId: projectId,
+          successful: successful,
+          failed: failed,
+          unknown: unknown,
+          inProgress: inProgress,
+          totalDuration: totalDuration,
+          day: day,
+        );
+
+        expect(buildDay.projectId, equals(projectId));
+        expect(buildDay.successful, equals(successful));
+        expect(buildDay.failed, equals(failed));
+        expect(buildDay.unknown, equals(unknown));
+        expect(buildDay.inProgress, equals(inProgress));
+        expect(buildDay.totalDuration, equals(totalDuration));
+        expect(buildDay.day, equals(day));
+      },
+    );
+
+    test(
+      "equals to another BuildDayData instance with the same parameters",
+      () {
+        final buildDayData = BuildDayData(
+          projectId: projectId,
+          successful: successful,
+          failed: failed,
+          unknown: unknown,
+          inProgress: inProgress,
+          totalDuration: totalDuration,
+          day: day,
+        );
+
+        final anotherBuildDayData = BuildDayData(
+          projectId: projectId,
+          successful: successful,
+          failed: failed,
+          unknown: unknown,
+          inProgress: inProgress,
+          totalDuration: totalDuration,
+          day: day,
+        );
+
+        expect(buildDayData, equals(anotherBuildDayData));
+      },
+    );
+
+    test(
+      ".fromJson() creates an instance from the given json",
+      () {
+        final expectedBuildDayData = BuildDayData(
+          projectId: projectId,
+          successful: successful,
+          failed: failed,
+          unknown: unknown,
+          inProgress: inProgress,
+          totalDuration: totalDuration,
+          day: day,
+        );
+
+        final json = createBuildDayDataJson(
+          projectId: projectId,
+          successful: successful,
+          failed: failed,
+          unknown: unknown,
+          inProgress: inProgress,
+          totalDuration: totalDuration.inMilliseconds,
+          day: Timestamp.fromDate(day),
+        );
+
+        final buildDayData = BuildDayData.fromJson(json);
+
+        expect(buildDayData, equals(expectedBuildDayData));
+      },
+    );
+
+    test(
+      ".fromJson() maps the null number of successful builds in the given json to zero",
+      () {
+        const expectedNumberOfBuilds = 0;
+        final json = createBuildDayDataJson(successful: null);
+
+        final buildDayData = BuildDayData.fromJson(json);
+
+        expect(buildDayData.successful, equals(expectedNumberOfBuilds));
```

Comment:
We can use the `isZero` matcher here.
```suggestion
        expect(buildDayData.successful, equals(isZero));
```